### PR TITLE
Add support to envvars with jobs

### PR DESCRIPTION
**Pull request by wpjunior**

### CODE REVIEW

**Review by crgisch — APPROVED** [review]
LGTM

**Comment by nettoclaudio** [comment]
So, I think we should follow the pattern used on app resource to set/unset env vars:
* POST on `/jobs/:name/env` to set them
* DELETE on `/jobs/:name/env` to unset them

**Comment by wpjunior** [comment]
@nettoclaudio do you think to also reuse same CLI commands ?

**Comment by nettoclaudio** [comment]
I think so. This way we can ensure fine-grained permissions (e.g. creating an user that can only change set/unset env vars), keep the pattern of app's commands (since users are already familiar with), keep the pattern of Terraform resources and so on.

File: api/app.go [hunk 1]
```diff
@@ -1200,16 +1200,25 @@ func setEnv(w http.ResponseWriter, r *http.Request, t auth.Token) (err error) {
 }
 
 func IsEnvVarValid(envs ...apiTypes.Env) error {
```
Comment (arthurcgc) [inline api/app.go:1202]: I would change the name to: ValidateApiEnvVars

Comment (arthurcgc) [inline api/app.go:1202]: why not just ? `envs []apiTypes.Env`

File: api/app.go [hunk 2]
```diff
@@ -1200,16 +1200,25 @@ func setEnv(w http.ResponseWriter, r *http.Request, t auth.Token) (err error) {
...
+	return isEnvsNamesVarValid(envNames...)
+}
+
+func isEnvsNamesVarValid(envs ...string) error {
 	var errs errors.MultiError
 
 	for _, e := range envs {
-		if isInternalEnv(e.Name) {
-			errs.Add(fmt.Errorf("cannot change an internal environment variable (%s): %w", e.Name, apiTypes.ErrWriteProtectedEnvVar))
+		if isInternalEnv(e) {
```
Comment (arthurcgc) [inline api/app.go:1215]: I would extract it to a function
```suggestion
func isValid(env string) error {
	if isInternalEnv(env) {
		return fmt.Errorf("cannot change an internal environment variable (%s): %w", env, apiTypes.ErrWriteProtectedEnvVar)
	}

	if err := isEnvVarName(env); err != nil {
		return fmt.Errorf("%q is not valid environment variable name: %w", env, err)
	}

	return nil
}
```

Comment (arthurcgc) [reply api/app.go:1215]: I would also change the name isEnvVarName to isUnixLike

**Review by nettoclaudio — APPROVED** [review]
lgtm

**Pull request merged**